### ISSUE TITLE
JedisClusterInfoCache renewClusterSlots() is not thread-safe

### ISSUE THREAD

**Issue by kutzi** [xetorthio/jedis#1854]
### Expected behavior

correct double-checked locking used on renewClusterSlots()

### Actual behavior

In rare conditions renewClusterSlots() could be executed multiple times, if several threads are triggering it at almost the same time.

### Steps to reproduce:

couldn't reproduce

#### Jedis version:
2.9.0

Description:
since the condition `if (!rediscovering)` is not checked again inside the locked block, it's not totally thread-safe.
It should actually look like this:
```
if (!rediscovering) {
      try {
        w.lock();
        if (!rediscovering) {
           rediscovering = true;
           ...
```

**Comment by kutzi** [comment]
See https://github.com/xetorthio/jedis/pull/1856

**Comment by sazzad16** [comment]
Closed by #1856
